streaming/src/main/scala/io/gearpump/streaming/task/Subscription.scala

```diff
@@ -189,6 +185,30 @@ class Subscription(
     maxPendingCount < maxPendingMessageCount
   }
 
+  def sendAckRequestToStallingTask(stallingTime: TimeStamp): Unit = {
+    minClockValue.indices.foreach { i =>
+      if (minClockValue(i) == stallingTime && allowSendingMoreMessages) {
+        updateMessageCount(i)
+        sendAckRequest(i, countIn = true)
+      }
+    }
+  }
+
+  private def sendAckRequest(partition: Int, countIn: Boolean = false): Unit = {
+    val targetTask = TaskId(processorId, partition)
+    val ackRequest = AckRequest(taskId, messageCount(partition), sessionId, countIn)
+    LOG.debug(s"sending ack request $ackRequest to $targetTask")
```

Comment:
This is on a critical path. 
LOG.debug can hurt your performance in unexpected way.